webpack/scenes/ContentViews/Table/tableDataGenerator.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { compoundExpand } from '@patternfly/react-table';
+import { ScreenIcon, FolderOpenIcon, ContainerNodeIcon } from '@patternfly/react-icons';
+
+import DetailsExpansion from '../expansions/DetailsExpansion';
+import RepositoriesExpansion from '../expansions/RepositoriesExpansion';
+import EnvironmentsExpansion from '../expansions/EnvironmentsExpansion';
+import VersionsExpansion from '../expansions/VersionsExpansion';
+import ContentViewName from '../components/ContentViewName';
+
+export const buildColumns = () => [
+  'Name', 'Last Published', 'Details',
+  { title: 'Environments', cellTransforms: [compoundExpand] },
+  { title: 'Repositories', cellTransforms: [compoundExpand] },
+  { title: 'Versions', cellTransforms: [compoundExpand] },
+];
+
+const buildRow = (contentView, openColumn) => {
+  const {
+    id, composite, name, environments, repositories, versions, last_published: lastPublished,
+  } = contentView;
+  const row = [
+    { title: <ContentViewName composite={composite ? 1 : undefined} name={name} cvId={id} /> },
+    lastPublished,
+    { title: 'Details', props: { isOpen: false, ariaControls: `cv-details-expansion-${id}`, contentviewid: id } },
```

Comment:
The `contentviewid` here (lowercase so it is recognized by React as a DOM attribute and not a prop) is so we have the content view id later on when we are accessing the expandables in the row or when the "details" link is working. Both need the Content view details endpoint, so we need the id to make the API call, somehow that information needs to be in the row.

Reply:
This is the error from the console when its `contentViewId` which is why I switched to `contentviewid`:

```
react_devtools_backend.js:6 Warning: React does not recognize the `contentViewId` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `contentviewid` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
```
Any ideas how to avoid that?

Reply:
@johnpmitsch Hmm, I didn't get that console error.  It's usually caused by using props spread syntax
```js
<MyComponent
  prop1={value}
  prop2={value}
  { ...props }
/>
```
which passes through whatever other props were there.  If props spread is used in an HTML element in JSX, you'll get that error.  In my experience it's usually a library that's doing it, so in this case probably a Patternfly component.

Reply:
I think the best way to avoid it is to avoid assigning arbitrary props to PF components (in other words, only assign props that are in the PF docs).  But that means you might have to find another place to put that data.

Reply:
Just realized the error you got was in `react_devtools_backend.js` so maybe it was React Dev Tools.

Reply:
still not sure how to go about this differently, but will keep looking